File: lua/pulpero/core/managers/tool/get_file.lua
local Tool = require("tool")
local OSCommands = require('OSCommands')

local function get_file(logger)
    local function execute(params)
        if not params.path then
            return { error = "Path is required" }
        end

        logger:debug("Looking for file content " .. params.path)
        local content = OSCommands:get_file_content(params.path)

        return { content = content }
    end

    return Tool.new(
        "get_file",
        "Get the content of a file by path",
        {
            path = {
                type = "string",
                description = "The file path"
            }
        },
        execute
    )
end

return {
    create_get_file_tool = get_file,
}

File: core/managers/tool/get_file.py
from core.managers.tool.tool import ToolResult, Tool
from core.util.logger import Logger

def get_file(logger: Logger) -> Tool:
    def execute(params) -> ToolResult:
        if params["path"] == None:
            return ToolResult(False, '', Exception("Path is required"))

        logger.debug(f"Looking for file content {params['path']}")
        response: str = ""

        try:
            with open(params["path"], 'r', encoding='utf-8') as f:
                response = f.read()
                f.close()
            return ToolResult(True,  response, None)
        except Exception as e:
            return ToolResult(False, '', e)

    return Tool(
        "get_file",
        "Get the content of a file by path",
        {
            'path': {
                'type': "string",
                'title': "path",
                'description': "The file path"
            }
        },
        ['path'],
        execute
    )
